bootstrap/pkg/kfapp/gcp/gcp.go

```diff
@@ -664,23 +664,29 @@ func (gcp *Gcp) updateDM(resources kftypes.ResourceEnum) error {
 		}
 	}
 
-	cluster, err := utils.GetClusterInfo(ctx, gcp.Spec.Project,
-		gcp.Spec.Zone, gcp.Name, gcp.tokenSource)
-	if err != nil {
-		return &kfapis.KfError{
-			Code: err.(*kfapis.KfError).Code,
-			Message: fmt.Sprintf("Configure K8s is failed: %v",
-				err.(*kfapis.KfError).Message),
+	// kfctl only
+	// Setup kube config
+	if gcp.isCLI {
+		// TODO(#2604): Need to create a named context.
+		credCmd := exec.Command("gcloud", "container", "clusters", "get-credentials",
+			gcp.Name,
+			"--zone="+gcp.Spec.Zone,
+			"--project="+gcp.Spec.Project)
+		credCmd.Stdout = os.Stdout
+		log.Infof("Running get-credentials %v --zone=%v --project=%v ...", gcp.KfDef.Name,
+			gcp.KfDef.Spec.Zone, gcp.KfDef.Spec.Project)
+		if err := credCmd.Run(); err != nil {
+			return &kfapis.KfError{
+				Code:    int(kfapis.INVALID_ARGUMENT),
+				Message: fmt.Sprintf("Error when running gcloud container clusters get-credentials: %v", err),
+			}
 		}
-	}
-	client, err := utils.BuildConfigFromClusterInfo(ctx, cluster, gcp.tokenSource)
-	if err != nil {
-		return &kfapis.KfError{
-			Code: err.(*kfapis.KfError).Code,
-			Message: fmt.Sprintf("Build ClientConfig error: %v",
-				err.(*kfapis.KfError).Message),
+		if _, err := os.Stat(kftypes.KubeConfigPath()); !os.IsNotExist(err) {
+			gcp.AddNamedContext()
 		}
 	}
+	// Get client for kube config.
+	client := kftypes.GetConfig()
```

Comment:
could you see if this method can be used?
https://github.com/kubeflow/kubeflow/blob/de1fc808b40f58e4bbb7199fedb68df29d412c98/bootstrap/pkg/utils/gcputils.go#L80

ideally we should isolate between GCP and k8s.

Reply:
That creates clientcmdapi.Config, but we need rest.Config

Reply:
then it's this one:
https://github.com/kubeflow/kubeflow/blob/de1fc808b40f58e4bbb7199fedb68df29d412c98/bootstrap/pkg/utils/gcputils.go#L60